wp_mobile_cache/src/repository/posts_transaction_tests.rs

```diff
@@ -0,0 +1,186 @@
+//! Transaction rollback tests for PostRepository.
+//!
+//! These tests verify that transaction failures properly rollback database state
+//! without leaving partial writes or corrupted data.
```

Comment:
This comment doesn't seem entirely correct as there are successful test cases as well.